src/main/perf/OrdinalMapBenchmark.java

```diff
@@ -0,0 +1,153 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package perf;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.LineNumberReader;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Locale;
+
+import org.apache.lucene.codecs.lucene91.Lucene91Codec;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.Field;
+import org.apache.lucene.document.SortedDocValuesField;
+import org.apache.lucene.index.DirectoryReader;
+import org.apache.lucene.index.DocValues;
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.index.IndexWriter;
+import org.apache.lucene.index.IndexWriterConfig;
+import org.apache.lucene.index.IndexWriterConfig.OpenMode;
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.index.LogDocMergePolicy;
+import org.apache.lucene.index.OrdinalMap;
+import org.apache.lucene.index.SerialMergeScheduler;
+import org.apache.lucene.index.SortedDocValues;
+import org.apache.lucene.store.FSDirectory;
+import org.apache.lucene.util.BytesRef;
+import org.apache.lucene.util.IOUtils;
+
+// javac -cp /l/trunk/lucene/core/build/libs/lucene-core-9.0.0-SNAPSHOT.jar src/main/perf/OrdinalMapBenchmark.java; java -cp /l/trunk/lucene/core/build/libs/lucene-core-9.0.0-SNAPSHOT.jar:. src/main/perf/OrdinalMapBenchmark.java /lucenedata/geonames/geonames.20160818.csv /l/indices/geonames -1
+
+/** Benchmark indexing stored fields on 1M lines of Geonames. */
+public class OrdinalMapBenchmark {
+
+  public static void main(String args[]) throws Exception {
+    if (args.length != 3) {
+      System.err.println("Usage: OrdinalMapBenchmark /path/to/geonames.txt /path/to/index/dir doc_limit(or -1 means index all lines)");
+      System.exit(1);
+    }
+
+    String geonamesDataPath = args[0];
+    String indexPath = args[1];
+    int docLimit = Integer.parseInt(args[2]);
+    
+    IOUtils.rm(Paths.get(indexPath));
+    try (FSDirectory dir = FSDirectory.open(Paths.get(indexPath))) {
+
+      System.err.println("Start indexing");
+      try (IndexWriter iw = new IndexWriter(dir, getConfig());
+          LineNumberReader reader = new LineNumberReader(new InputStreamReader(Files.newInputStream(Paths.get(geonamesDataPath))))) {
+        indexDocs(iw, reader, docLimit);
+      }
+
+      try (DirectoryReader reader = DirectoryReader.open(dir)) {
+        System.err.println("Warm up");
+        for (String field : new String[] { "id", "name", "country_code", "time_zone" }) {
+          loadOrdinalMap(reader, field);
+        }
+
+        System.err.println("Now run benchmark");
+        for (String field : new String[] { "id", "name", "country_code", "time_zone" }) {
+          // Take the min across multiple runs to decrease noise
```

Comment:
I don't understand this comment, honestly don't think taking minimum decreases noise. I'm aware luceneutil does this everywhere, I just want to say I don't think it works. To decrease noise, we have to write benchmarks that aren't noisy :)

Reply:
OK I removed this.